### PR TITLE
Set up for Mongo database and further docker composition improvements

### PR DESCRIPTION
**Pull request by Adimaano**
So we are still missing a database to have a "full stack" of our little dynamic web app.

We are choosing MongoDB cause 
1. I never used it and 
2. It's supposed to be really simple and easy to use!

And can you believe it. It is TRIVIAL to set it up. The official docker images are doing all the work.

So really its just adding dockerfile and composing it.
Much more interesting is the database "management".

There are multiple options on how to access the database and manage the data.
The most intuitive one's of course use a dashboard. In most productive cases there is ALWAYS a dashboard because you have to generally work with **millions** of data.

All options on how to get a dashboard to display the data and browse them properly are also named in the readme too, but here they are:

1) Use your own / any local database browser:
Download and Install any Database Browser on your PC and connect it to our hosted MongoDB container.
The official one by MongoDB is "Compass".
There is also an vscode extension, which can connect and browse too. (no dashboard though, very code-y interface)

2) Spin up your own MongoDB GUI !:
MongoDB also has the "mongo express" image. You can add this to the compose file to spin up ANOTHER service / container.
This one is its own web application, that you can host yourself. It connects to our mongo database too and you can browse it on that web application's port.
(On the job, this is usually what is done)

In the spirit of experimentation and to not force a method, I choose to provide neither option here.
I use the vscode extension to browse the mongodb.
See what you like best and understand why and how a database works / is setup in a project.

Post-Review:
Passing Docker environment variables has niche quirks, but works fine.
Not 100% sure where the correct location to store and define them all. 
Each for container level one? ( via defined env_file attribute )
Compose level one? ( via default .env ) 
Decided on .env for simplicity and because we don't need more yet.

Also take specific care over PRECEDENCE:
https://docs.docker.com/compose/environment-variables/envvars-precedence/

Init Credentials are not encrypted. For PoC sake this should be enough to start coding! :)
In the future we will be setting up new admin and user credentials for database management. Those shall be encrypted and saved as "secrets"!
https://docs.docker.com/engine/swarm/secrets/

Any takers? :D

### CODE REVIEW

File: compose.yaml [hunk 1]
```diff
@@ -18,6 +18,21 @@ services:
       context: ./frontend/
     ports:
       - 5001:8080
+  mongo:
+    build:
+      context: ./database/
+    ports:
+      - 5002:27017
```
Comment (buenaflor) [inline compose.yaml]:
```suggestion
      - "${MONGO_PORT}:27017"
```

ports should generally not be hardcoded

File: database/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Mongo image docker pulls from the official repos ( 7.0 is most recent release )
+FROM mongo:7.0
+
+# Setting default credentials to access mongodb
+ENV MONGO_INITDB_ROOT_USERNAME=mongoadmin
+ENV MONGO_INITDB_ROOT_PASSWORD=mongopass
```
Comment (buenaflor) [inline database/Dockerfile]:
```suggestion
```

guess we don't need these anymore if we pass them through docker-compose config

File: compose.yaml (hunk 1 above)
Comment (buenaflor) [reply compose.yaml]: also relevant for the other services, frontend & backend

File: compose.yaml [hunk 3]
```diff
@@ -18,6 +18,21 @@ services:
       context: ./frontend/
     ports:
       - 5001:8080
+  mongo:
+    build:
```
Comment (buenaflor) [inline compose.yaml:13]:
```suggestion
  mongo:
    env_file: "mongodb.env"
    build:
```
```env
# mongodb.env example
MONGO_INITDB_ROOT_USERNAME=mongoadmin
MONGO_INITDB_ROOT_PASSWORD=mongopass
MONGO_PORT=5002
```

we can start using env for making this less prone to mistakes, the example .env file can look like this so everyone knows they have to add the values for local development

```env
# For local development add the missing config
MONGO_INITDB_ROOT_USERNAME=
MONGO_INITDB_ROOT_PASSWORD=
MONGO_PORT=
```

**Review by buenaflor — COMMENTED** [review]
added some comments regarding env variables, generally should be good to introduce it early in the process to minimize mistakes later on

I'm not a docker expert so syntax might be off

File: compose.yaml (hunk 3 above)
Comment (Adimaano) [reply compose.yaml:13]: env_file is a nice idea yeah. I'll use a general one to use as config file with all ports as you also suggest.

Comment (Adimaano) [reply compose.yaml:22]: I'll keep the environment variables as arguments in the Dockerfiles as container "default" values.

According to official docs, env file takes precedence and overwrites those if any dev changes those.
https://docs.docker.com/compose/environment-variables/envvars-precedence/

File: compose.yaml [hunk 4]
```diff
@@ -1,53 +1,28 @@
...
+      context: ./database/
+    command: bash -c "mongod | jq --unbuffered -r '.msg'" 
+    ports:
+      - ${HOSTPORT_MONGO}:27017
+    volumes:
+      - mongodata:/data/db
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME=${INIT_ADMIN}
+      - MONGO_INITDB_ROOT_PASSWORD=${INIT_PASS}
```
Comment (buenaflor) [inline compose.yaml]: if we specify the key to be `MONGO_INITDB_ROOT_USERNAME` and `MONGO_INITDB_ROOT_PASSWORD` in the .env file we don't need to additionally add this config

Comment (buenaflor) [reply compose.yaml]: unless you wanna reuse it for other dbs then its fine to keep it

Comment (Adimaano) [reply compose.yaml]: Thought about reusing it, but alas; Its just the init credentials!

Done:
Moved it over to .env

PS.: When we use actual DB creds, we should implement secrets. For PoC sake this works for now.

File: database/Dockerfile (hunk 2 above)
Comment (Adimaano) [reply database/Dockerfile]: Done:

passing through .env

File: compose.yaml (hunk 1 above)
Comment (Adimaano) [reply compose.yaml]: Done:

passing through .env

**Review by buenaflor — APPROVED** [review]
👍

**Pull request merged**